community/planning/batch_processor.md

```diff
@@ -0,0 +1,84 @@
+<!--
+  Copyright 2021 Cargill Incorporated
+  Licensed under Creative Commons Attribution 4.0 International License
+  https://creativecommons.org/licenses/by/4.0/
+-->
+
+# Batch Processor
+
+The batch processor is a proposed component meant to monitor a database that
+holds batched transactions and send them to either a Hyperledger Sawtooth or
+Splinter network. Batch processors are meant to be run as part of a cluster with
+other batch processors in order to provide redundancy or reduce the time it
+takes for batches to be processed.
+
+## Batch Submission Algorithm
```

Comment:
This algorithm, without some form of locking, would seem to suffer from multiple processes easily claiming the same batches due to the race condition between the SELECT and implied UPDATE. Thus more detail here around the specific mechanics would be useful.

Reply:
Updated to specify that a submitter is claiming a lock and that the update and query for batches are a part of the same transaction.